modules/codegen/src/main/scala/com/twilio/guardrail/extract/package.scala

```diff
@@ -1,22 +1,38 @@
 package com.twilio.guardrail
 
 package object extract {
-  def ScalaTracingLabel[F: VendorExtension.VendorExtensible](v: F): Option[String] =
-    VendorExtension(v).extract[String]("x-scala-tracing-label")
-  def ScalaPackage[F: VendorExtension.VendorExtensible](v: F): Option[String] =
-    VendorExtension(v).extract[String]("x-scala-package")
-  def ScalaType[F: VendorExtension.VendorExtensible](v: F): Option[String] =
-    VendorExtension(v).extract[String]("x-scala-type")
+  private def extractFromNames[F: VendorExtension.VendorExtensible, T: Extractable](v: F, names: List[String]): Option[T] =
+    names.foldLeft(Option.empty[T])(
```

Comment:
This is actually `names.collectFirstSome(VendorExtension(v).extract[T](_))`, fwiw

Reply:
Ooh, TIL.  Will update and then merge.